### PR TITLE
docs: add reference hardware documentation

### PR DESCRIPTION
**Pull request by NilaySener**
## Description
The Autoware documentation has been updated with a new page titled "_Reference HW_ " which includes information on various hardware components. This pull request aims to incorporate these additions into the documentation. The added content covers AD computers, LiDARs, RADARs, CAMERAs, thermal CAMERAs, IMU, AHRS & GNSS/INS sensors, vehicle drive-by-wire suppliers, vehicle platform suppliers, remote drive solutions, and a full drivers list. This PR enhances the documentation by providing valuable information about these hardware options for Autoware users.

The referenced HW document is [here.](https://docs.google.com/document/d/1Q2hRkJ1y_smtogIYb0sQdOyot-01UpZV3DlGyGU1-Gs/edit)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
Can you also share the original HW document link in the PR description?

File: docs/reference-hw/lidars.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Link to company website:  
+[http://www.lslidar.com/](http://www.lslidar.com/)
+
+## **Livox 3D LIDAR Sensors**
+
+Livox Lidars which has ROS2 driver and tested by one or more community members are listed below:
+
+| Supported Products List | Range | FOV (V), (H)      | ROS2 Driver | Autoware Tested (Y/N) |
+| ----------------------- | ----- | ----------------- | ----------- | --------------------- | --- |
+| Horizon                 | 260m  | (81.7°), (25.1°)  | Y           | Y                     | -   |
```
Comment (mitsudome-r) [inline docs/reference-hw/lidars.md:78]: Table is corrupted for Livox Lidars
```suggestion
| ----------------------- | ----- | ----------------- | ----------- | --------------------- |
| Horizon                 | 260m  | (81.7°), (25.1°)  | Y           | Y                     |
```

File: docs/reference-hw/lidars.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+| Supported Products List | Range | FOV (V), (H)          | ROS2 Driver | Autoware Tested (Y/N) |
+| ----------------------- | ----- | --------------------- | ----------- | --------------------- |
+| Alpha Prime             | 245m  | (+15°)/(-25°), (360°) | Y           | Y                     |
+| Ultra Puck              | 200m  | (+15°)/(-25°), (360°) | Y           | Y                     |
+| Puck                    | 100m  | (+15°)/(-15°), (360°) | Y           | Y                     |
+| Puck Hi-res             | 100m  | (+10°)/(-10°), (360°) | Y           | Y                     |
+
+Link to ROS2 drivers:  
+ [https://github.com/ros-drivers/velodyne/tree/ros2/velodyne_pointcloud](https://github.com/ros-drivers/velodyne/tree/ros2/velodyne_pointcloud)  
+ [https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/tree/master/src/drivers/velodyne_nodes](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/tree/master/src/drivers/velodyne_nodes)
```
Comment (mitsudome-r) [inline docs/reference-hw/lidars.md:16]: Maybe it's better to put AWF driver as well.
https://github.com/autowarefoundation/awf_velodyne/tree/tier4/universe

File: docs/reference-hw/remote_drive.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Link to company website:  
+[https://fortrobotics.com/vehicle-safety-controller/](https://fortrobotics.com/vehicle-safety-controller/)
+
+## **LOGITECH**
+
+Logitech joysticks which are used for autonomous driving and tested by one or more community members are listed below:
+
+| Supported Products | Op.Frequency                | Controller               | ROS2 Support | Autoware Tested (Y/N) |
+| ------------------ | --------------------------- | ------------------------ | ------------ | --------------------- | --- |
+| Logitech F-710     | 2.4 GHz Wireless, 10m range | (2) 2-axis joysticks<br> | (18) buttons | Y                     | Y   |
```
Comment (mitsudome-r) [inline docs/reference-hw/remote_drive.md:20]:
```suggestion
| Supported Products | Op.Frequency                | Controller               | ROS2 Support | Autoware Tested (Y/N) |
| ------------------ | --------------------------- | ------------------------ | ------------ | --------------------- |
| Logitech F-710     | 2.4 GHz Wireless, 10m range | (2) 2-axis joysticks<br> | (18) buttons | Y                     | Y   |
```

**Review by mitsudome-r — COMMENTED** [review]
Thanks for making the PR. I made some review comments.

**Review by mitsudome-r — APPROVED** [review]
LGTM

**Pull request merged**